pkg/version/channelversion.go

```diff
@@ -12,20 +13,43 @@ type channelVersion struct {
 	version string
```

Comment:
rather than encoding the hotpatch and build into into the version field and having a "Hotpatch" suffix as a marker on the channel, can we parse the channel version into 4 distinct fields: channel, version, build, hotpatch?  then we can define a comparison function for channelVersion which compares channel, version, and hotpatch, but does not consider build.